ksql-metastore/src/main/java/io/confluent/ksql/metastore/model/StructuredDataSource.java

```diff
@@ -60,6 +60,10 @@
     this.ksqlTopic = requireNonNull(ksqlTopic, "ksqlTopic");
     this.keySerde = requireNonNull(keySerde, "keySerde");
     this.serdeOptions = ImmutableSet.copyOf(requireNonNull(serdeOptions, "serdeOptions"));
+
+    if (!schema.withImplicitFields().equals(schema)) {
+      throw new IllegalArgumentException();
```

Comment:
let's add a message into here

Reply:
This is just temporary. It's changed in later PR.